test/unit_tests/test_translation.py

```diff
@@ -541,6 +568,95 @@ def amplitude(p, t):
     assert np.allclose(waveforms[0].amplitudes, amplitudes)
 
 
+def test_translate_parametrized_evolution_mixed():
+    """Test that a ParametrizedEvolution with one constant and one callable amplitude pulse
+    is translated to a PulseGate correctly."""
+    n_wires = 4
+    dev = _aws_device(wires=n_wires, device_arn="arn:aws:braket:eu-west-2::device/qpu/oqc/Lucy")
+
+    class DummyProperties:
+        def __init__(self):
+            self.pulse = PulseDeviceActionProperties.parse_raw(OQC_PULSE_PROPERTIES)
+            self.paradigm = GateModelQpuParadigmProperties.parse_raw(OQC_PARADIGM_PROPERTIES)
+
+    dev._device._properties = DummyProperties()
+
+    H = transmon_drive(0.02, np.pi, 0.5, [0])
+    H += transmon_drive(amplitude, -np.pi / 2, 0.75, [1])
+
+    amplitude_param = 0.1
+    op = ParametrizedEvolution(H, [amplitude_param], t=50)
+
+    braket_gate = translate_operation(op, device=dev)
+
+    assert isinstance(braket_gate, gates.PulseGate)
+    assert braket_gate.qubit_count == 2
+
+    ps = braket_gate.pulse_sequence
+    expected_frames = [dev._device.frames[f"q{w}_drive"] for w in range(2)]
+
+    frames = list(ps._frames.values())
+    assert len(frames) == 2
+    assert all(actual == expected for actual, expected in zip(frames, expected_frames))
+
+    dt = expected_frames[0].port.dt * 1e9
+    amplitudes = [amplitude(amplitude_param, t) for t in np.arange(0, 50 + dt, dt)]
+
+    waveforms = list(ps._waveforms.values())
```

Comment:
Knowing `ps._waveforms` was accessible like this would have saved past-me a lot of time 😆